### PR TITLE
📖 Add trigger phrases and status badge for e2e tests

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

This adds the trigger phrases for the e2e tests to the README.md and also the build badge for the periodic job.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

**Comment by maxrantil** [comment]
nice, LGTM

**Comment by lentzi90** [comment]
Wait there is something wrong with the build badge. It doesn't show the correct text
/hold

**Comment by lentzi90** [comment]
Ok now it looks better
/unhold

**Review by kashifest — COMMENTED** [review]
/approve

File: README.md [hunk 1]
```diff
@@ -39,6 +40,11 @@ Eventually these tests will be replaced with BMO's own
 [e2e tests](test/e2e/README.md). BMO e2e tests are currently under active
 development. Here are the tests which run on different branches in BMO:
 
+**E2e tests:**
+
+* **/test metal3-bmo-e2e-test-pull** runs required BMO e2e tests on your PR. Works for main and release-0.5 branches.
```
Comment (elfosardo) [inline README.md:45]: would be interesting to have a description of the tests included here and the optional ones too

Comment (lentzi90) [reply README.md:45]: Good point! I wonder if it perhaps is a bit premature, at least for the optional tests? I'm afraid the list would be constantly changing (or worse, outdated).
Another question is if it should be here or in the `test/e2e/README.md`?
What do you think?

Comment (elfosardo) [reply README.md:45]: good point for the optional tests, let's wait a bit for when they're more "stable"
I guess adding the description for the required ones to `test/e2e/README.md` and add a link would work

File: README.md [hunk 2]
```diff
@@ -39,6 +40,13 @@ Eventually these tests will be replaced with BMO's own
 [e2e tests](test/e2e/README.md). BMO e2e tests are currently under active
 development. Here are the tests which run on different branches in BMO:
 
+**E2e tests:**
```
Comment (tuminoid) [inline README.md:43]:
```suggestion
**E2E tests:**
```

**Review by tuminoid — COMMENTED** [review]
One nit.

**Comment by tuminoid** [comment]
LGTM, I'll leave it to @elfosardo to flag it if he's also happy.

File: test/e2e/README.md [hunk 3]
```diff
@@ -34,13 +35,44 @@ export GINKGO_FOCUS="upgrade"
...
 
 ```bash
 make test-e2e
 ```
+
+## Tests
+
+Here is a list of the currently implemented tests. As mentioned above, we divide
+them into two sets, one required and one optional. The required tests must pass
+on every PR before merging. Optional tests may be triggerd for extra
```
Comment (elfosardo) [inline test/e2e/README.md:53]: nit: triggered

**Review by elfosardo — COMMENTED** [review]
/lgtm

**Pull request merged**